### DISCUSSION TITLE
Error converting text with font to image

### DISCUSSION THREAD

**Discussion by lucianozr** [ImageMagick/ImageMagick#5737] [Help]
Hello good afternoon.
I have an editor that the user can create templates, edit layers and add text with stylized fonts.
I recently came across a big problem that I've been investigating for a long time.

When transforming text with stylized font into an image and trying to place this new image in the template, it gets misaligned.
Below is the code where this operation takes place.

Does anyone have an idea what I could be doing wrong?

execSync(`convert  ${font ? `-font '${font}'` : ''} -background none -evaluate multiply ${layer.opacity / 100} -fill '${layer.color}' -gravity ${layer.textAlign ? alignTexts[layer.textAlign] : alignTexts.left}  -pointsize ${layer.fontSize} -size ${layer.size.width}x caption:'${layer.value}' /tmp/${layer.id}.png`);

commandString += ` -draw "image srcOver ${layer.position.x},${layer.position.y} 0,0 '/tmp/${layer.id}.png' "

**Comment by fmw42** [comment]
Reset your gravity setting after using it back to the default before caption:  Use either +gravity or -gravity northwest.
